src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCheck.java

```diff
@@ -215,12 +215,11 @@ public void visitToken(DetailAST ast) {
 
         if ((before == -1 || CommonUtil.isCodePointWhitespace(line, before))
                 && !isInEmptyForInitializerOrCondition(ast)) {
-            boolean flag = !allowLineBreaks;
+            final boolean hasNoWhitespaceBefore = !allowLineBreaks
+                || before > -1
+                && !CodePointUtil.hasWhitespaceBefore(before, line);
             // verify all characters before '.' are whitespace
```

Comment:
I think we can remove comment now, our intention is very clear.

Reply:
done

Reply:
Resolved.